src/herder/TxSetFrame.cpp

```diff
@@ -806,9 +806,11 @@ makeTxSetFromTransactions(PerPhaseTransactionList const& txPhases,
         }
     }
 
+    // We already trimmed invalid transactions in an earlier call to
+    // `trimInvalid`, so skip validation here
     valid = valid &&
             outputApplicableTxSet->checkValid(app, lowerBoundCloseTimeOffset,
-                                              upperBoundCloseTimeOffset);
+                                              upperBoundCloseTimeOffset, true);
```

Comment:
This looks like it checks out. applySurgePricing is the only suspicious looking function, but to double check my understanding, this just returns a set of txs that can be included after applying surge pricing, such that no previously validated tx could become invalidated due to an increase in base fee after calling this function?